File: Backend/TrackrAPI/TrackrAPI/Services/Contabilidad/AgrupadorCuentaContableService.cs
using TrackrAPI.Dtos.Contabilidad;
using TrackrAPI.Models;
using TrackrAPI.Repositorys.Contabilidad;
using System.Collections.Generic;

namespace TrackrAPI.Services.Contabilidad
{
    public class AgrupadorCuentaContableService
    {
        private IAgrupadorCuentaContableRepository agrupadorCuentaContableRepository;
        private AgrupadorCuentaContableValidatorService agrupadorCuentaContableValidatorService;

        public AgrupadorCuentaContableService(
            IAgrupadorCuentaContableRepository agrupadorCuentaContableRepository,
            AgrupadorCuentaContableValidatorService agrupadorCuentaContableValidatorService
        )
        {
            this.agrupadorCuentaContableRepository = agrupadorCuentaContableRepository;
            this.agrupadorCuentaContableValidatorService = agrupadorCuentaContableValidatorService;
        }

        public AgrupadorCuentaContableDto ConsultarDto(int idAgrupadorCuentaContable)
        {
            return agrupadorCuentaContableRepository.ConsultarDto(idAgrupadorCuentaContable);
        }

        public IEnumerable<AgrupadorCuentaContableDto> ConsultarParaGrid()
        {
            return agrupadorCuentaContableRepository.ConsultarParaGrid();
        }

        public IEnumerable<AgrupadorCuentaContableDto> ConsultarParaSelector()
        {
            return agrupadorCuentaContableRepository.ConsultarParaSelector();
        }

        public void Agregar(AgrupadorCuentaContableDto agrupadorDto)
        {
            AgrupadorCuentaContable agrupador = new AgrupadorCuentaContable()
            {
                Clave = agrupadorDto.Clave,
                Descripcion = agrupadorDto.Descripcion,
                IdCuentaContableCapital = agrupadorDto.IdCuentaContableCapital,
                IdCuentaContableResultado = agrupadorDto.IdCuentaContableResultado
            };

            agrupadorCuentaContableValidatorService.ValidarAgregar(agrupador);
            agrupadorCuentaContableRepository.Agregar(agrupador);
        }

        public void Editar(AgrupadorCuentaContableDto agrupadorDto)
        {
            AgrupadorCuentaContable agrupador = new AgrupadorCuentaContable()
            {
                IdAgrupadorCuentaContable = agrupadorDto.IdAgrupadorCuentaContable,
                Clave = agrupadorDto.Clave,
                Descripcion = agrupadorDto.Descripcion,
                IdCuentaContableCapital = agrupadorDto.IdCuentaContableCapital,
                IdCuentaContableResultado = agrupadorDto.IdCuentaContableResultado
            };

            agrupadorCuentaContableValidatorService.ValidarEditar(agrupador);
            agrupadorCuentaContableRepository.Editar(agrupador);
        }

        public void Eliminar(int idAgrupadorCuentaContable)
        {
            agrupadorCuentaContableValidatorService.ValidarEliminar(idAgrupadorCuentaContable);

            AgrupadorCuentaContable agrupador = agrupadorCuentaContableRepository.Consultar(idAgrupadorCuentaContable);
            agrupadorCuentaContableRepository.Eliminar(agrupador);
        }
    }
}

File: Frontend/trackr-administrador/src/app/shared/http/contabilidad/agrupador-cuenta-contable.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { AgrupadorCuentaContable } from '@models/contabilidad/agrupador-cuenta-contable';

@Injectable()
export class AgrupadorCuentaContableService {
  private dataUrl = 'agrupadorCuentaContable/';

  constructor(private http: HttpClient) { }

  public consultarParaGrid(): Observable<AgrupadorCuentaContable[]> {
    return this.http.get<AgrupadorCuentaContable[]>(this.dataUrl + `consultarParaGrid`);
  }

  public consultarParaSelector(): Observable<AgrupadorCuentaContable[]> {
    return this.http.get<AgrupadorCuentaContable[]>(this.dataUrl + `consultarParaSelector`);
  }

  public consultarDto(idAgrupador: number): Observable<AgrupadorCuentaContable> {
    return this.http.get<AgrupadorCuentaContable>(this.dataUrl + `consultarDto/${idAgrupador}`);
  }

  public agregar(agrupador: AgrupadorCuentaContable): Observable<void> {
    return this.http.post<void>(this.dataUrl + `agregar`, agrupador);
  }

  public editar(agrupador: AgrupadorCuentaContable): Observable<void> {
    return this.http.put<void>(this.dataUrl + `editar`, agrupador);
  }

  public eliminar(idAgrupador: number): Observable<void> {
    return this.http.delete<void>(this.dataUrl + `eliminar/${idAgrupador}`);
  }
}